[skia] build failure

<!-- ⚠ Please use the generated issue template instead to report a port build failure. ⚠ -->
<!-- ⁉ You can find its location at the end of the console output. ⁉ -->

**Host Environment**

- OS: Windows
- Compiler: VC2022

`./vcpkg install skia`

**Failure logs**

CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: D:/dev/test1/installed/x64-windows/tools/gn/gn.exe gen D:/dev/test1/buildtrees/skia/x86-windows-dbg "--args=target_cpu=\"x86\" target_os=\"win\" is_component_build=true skia_use_fontconfig=false skia_use_freetype=false skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=truetarget_cpu=\"x86\" target_os=\"win\" is_component_build=true skia_use_fontconfig=false skia_use_freetype=false skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=true skia_use_vulkan=true win_vc=\"C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\" skia_use_lua=false skia_enable_tools=false skia_enable_spirv_validation=false is_debug=true     extra_cflags_c=[\"-nologo\",\"-DWIN32\",\"-D_WINDOWS\",\"-W3\",\"-utf-8\",\"-MP\",\"-D_DEBUG\",\"-MDd\",\"-Z7\",\"-Ob0\",\"-Od\",\"-RTC1\"]     extra_cflags_cc=[\"-nologo\",\"-DWIN32\",\"-D_WINDOWS\",\"-W3\",\"-utf-8\",\"-GR\",\"-EHsc\",\"-MP\",\"-D_DEBUG\",\"-MDd\",\"-Z7\",\"-Ob0\",\"-Od\",\"-RTC1\"]"
    Working Directory: D:/dev/test1/buildtrees/skia/src/40cd0870fd-80923cf1f9.clean
    Error code: 1
    See logs for more information:
      D:\dev\test1\buildtrees\skia\generate-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-gn/vcpkg_gn_configure.cmake:11 (vcpkg_execute_required_process)
  installed/x64-windows/share/vcpkg-gn/vcpkg_gn_configure.cmake:35 (z_vcpkg_gn_configure_generate)
  ports/skia/portfile.cmake:271 (vcpkg_gn_configure)
  scripts/ports.cmake:147 (include)

From (D:\dev\test1\buildtrees\skia\generate-x86-windows-dbg-out.log):
ERROR at the command-line "--args":1:160: Operator requires a rvalue.
target_cpu="x86" target_os="win" is_component_build=true skia_use_fontconfig=false skia_use_freetype=false skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=truetarget_cpu="x86" target_os="win" is_component_build=true skia_use_fontconfig=false skia_use_freetype=false skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=true skia_use_vulkan=true win_vc="C:\Program Files\Microsoft Visual Studio\2022\Community\VC" skia_use_lua=false skia_enable_tools=false skia_enable_spirv_validation=false is_debug=true     extra_cflags_c=["-nologo","-DWIN32","-D_WINDOWS","-W3","-utf-8","-MP","-D_DEBUG","-MDd","-Z7","-Ob0","-Od","-RTC1"]     extra_cflags_cc=["-nologo","-DWIN32","-D_WINDOWS","-W3","-utf-8","-GR","-EHsc","-MP","-D_DEBUG","-MDd","-Z7","-Ob0","-Od","-RTC1"]
                                                                                                                                                               ^--------------------
This thing on the right does not evaluate to a value.

<!-- ‼ **The console output by itself is not enough to investigate a build failure.** ‼ -->

**Additional context**

Running it a second time gives weirder errors.  Seems pretty broken.

I can confirm the same thing happens to me on Ubuntu 20.04 with GCC 9.4.0. Output from `generate-x64-linux-dbg-out.log`:

```
ERROR at the command-line "--args":1:143: Operator requires a rvalue.
target_cpu="x64" is_component_build=false skia_use_fontconfig=true skia_use_freetype=true skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=truetarget_cpu="x64" is_component_build=false skia_use_fontconfig=true skia_use_freetype=true skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=true skia_use_vulkan=true         cc="/usr/bin/cc"         cxx="/usr/bin/c++" skia_use_lua=false skia_enable_tools=false skia_enable_spirv_validation=false is_debug=true     extra_cflags_c=["-fPIC","-g"]     extra_cflags_cc=["-fPIC","-g"]
                                                                                                                                              ^--------------------
This thing on the right does not evaluate to a value.
```

Output on the command line:

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /home/neuhauser/Programming/C++/vcpkg/installed/x64-linux/tools/gn/gn gen /home/neuhauser/Programming/C++/vcpkg/buildtrees/skia/x64-linux-dbg "--args=target_cpu=\"x64\" is_component_build=false skia_use_fontconfig=true skia_use_freetype=true skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=truetarget_cpu=\"x64\" is_component_build=false skia_use_fontconfig=true skia_use_freetype=true skia_use_harfbuzz=true skia_use_icu=true skia_use_gl=true skia_use_vulkan=true         cc=\"/usr/bin/cc\"         cxx=\"/usr/bin/c++\" skia_use_lua=false skia_enable_tools=false skia_enable_spirv_validation=false is_debug=true     extra_cflags_c=[\"-fPIC\",\"-g\"]     extra_cflags_cc=[\"-fPIC\",\"-g\"]"
```

Seems like unfortunately the build just now on Ubuntu only worked because I have vulkan-memory-allocator installed via the `vma` package provided by the Vulkan SDK PPA. When I try to compile it on Windows with MSVC (Vulkan SDK also installed on this system), I get the following error message.

```
C:\Users\neuhauser\Programming\vcpkg-dg0yt\buildtrees\skia\src\40cd0870fd-23740c221c.clean\third_party\vulkanmemoryallocator\GrVulkanMemoryAllocator.h(35): fatal error C1083: Cannot open include file: 'vk_mem_alloc.h': No such file or directory
```

Apparently, `third_party/externals/vulkanmemoryallocator` is missing from the vcpkg Skia buildtree. This directory gets created when I manually compile Skia without vcpkg (and contains `include/vk_mem_alloc.h`).

--- a/ports/skia/portfile.cmake
+++ b/ports/skia/portfile.cmake
@@ -178,7 +178,7 @@ if("metal" IN_LIST FEATURES)
 endif()
 
 if("vulkan" IN_LIST FEATURES)
-    string(APPEND OPTIONS "${OPTIONS} skia_use_vulkan=true")
+    string(APPEND OPTIONS " skia_use_vulkan=true")
 endif()
 
 if("direct3d" IN_LIST FEATURES)

--- a/ports/skia/portfile.cmake
+++ b/ports/skia/portfile.cmake
@@ -179,6 +179,7 @@ endif()
 
 if("vulkan" IN_LIST FEATURES)
     string(APPEND OPTIONS " skia_use_vulkan=true")
+    file(COPY "${CURRENT_INSTALLED_DIR}/include/vk_mem_alloc.h" DESTINATION "${SOURCE_PATH}/third_party/vulkanmemoryallocator")
 endif()
 
 if("direct3d" IN_LIST FEATURES)

--- a/ports/skia/vcpkg.json
+++ b/ports/skia/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "skia",
   "version": "0.36.0",
-  "port-version": 6,
+  "port-version": 7,
   "description": [
     "Skia is an open source 2D graphics library which provides common APIs that work across a variety of hardware and software platforms.",
     "It serves as the graphics engine for Google Chrome and Chrome OS, Android, Mozilla Firefox and Firefox OS, and many other products.",

--- a/ports/skia/vcpkg.json
+++ b/ports/skia/vcpkg.json
@@ -136,7 +136,8 @@
     "vulkan": {
       "description": "Vulkan support for skia",
       "dependencies": [
-        "vulkan"
+        "vulkan",
+        "vulkan-memory-allocator"
       ]
     }
   }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7170,7 +7170,7 @@
     },
     "skia": {
       "baseline": "0.36.0",
-      "port-version": 6
+      "port-version": 7
     },
     "skyr-url": {
       "baseline": "1.13.0",

--- a/versions/s-/skia.json
+++ b/versions/s-/skia.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "a358418f4df9a797c1e4a235ae12d191170af2b3",
+      "git-tree": "508710af38e0b50675532741bf899ec4c3f20c3b",
       "version": "0.36.0",
       "port-version": 7
     },

--- a/versions/s-/skia.json
+++ b/versions/s-/skia.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a358418f4df9a797c1e4a235ae12d191170af2b3",
+      "version": "0.36.0",
+      "port-version": 7
+    },
+    {
       "git-tree": "0458cf18e0a28b607b8d3a2ca9fba56395d5f147",
       "version": "0.36.0",
       "port-version": 6

